### DISCUSSION TITLE
Missing list of possible values of context, for example: `github.event_name`

### DISCUSSION THREAD

**Discussion by yeongrokgim** [community/community#110254] [Actions]
### Select Topic Area

Product Feedback

## TL;DR

Documentation is way too messy, different level of contents is mixed up. There should be seperate chapter for developer reference values.

## Context

I want to configure conditional workflows, something like:

```yaml
jobs:
  build-and-test:
    runs-on: ubuntu-latest
    steps:
      - name: Build something
        run: |
          echo "Build passed"
      - name: Test something before merge
        # Question:
        #  Will `merge_queue` works? not `merge_group`? how about `mergequeue`?
        #  It seems `pull_request` is valid value, but no concrete reference.
        if: ${{ github.event_name == 'merge_queue' }} 
        run: |
          echo "Test failed"
          exit 1
```

The documentation pretends it contains enough information, for example 
https://docs.github.com/en/actions/learn-github-actions/contexts#example-usage-of-the-github-context:~:text=webhook%20payload.-,github.event_name,-string

But actually this already too messy. Above link looks something simillar to `developer reference`, but the hierarchy is `GitHub Actions -> Learn GitHub Actions -> Contexts`. The name of the section `Learn GitHub Actions` clearly indicates this section contains introductory resources, not an exhaustive list of json structure.

Besides from [`learn-github-actions/contexts`](https://docs.github.com/en/actions/learn-github-actions/contexts) page, the list of `github.event_name` value for different triggers could be available from [`using-workflows/events-that-trigger-workflows`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows), but IT IS NOT.

The only available workaround is:

1) Run debug workflow myself, print every context with different combination of triggers and conditions
2) Reverse-engineer outputs and values
3) Pray for object structure have not been changed and not to be changed

## Suggestion

GitHub may want to deletgate docs maintanence to community users, but this is not the case for closed-source product like GitHub Actions.
Moreover, this could be too large to be updated by individual opensource contributors.

Here's my suggestion,

1. Refactor documentation to split `1) introductory resources (quick start and examples)` and `2) Developer reference with full list of values (every possible values for every context provided by Actions)`
2. Generate dev refs from actual running version of GitHub Actions source code.

## Links

- https://docs.github.com/en/actions/learn-github-actions/contexts
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

**Comment by jainaks01-bh** [comment]
type github.event_name = 'push' | 'workflow_dispatch'

**Comment by WillGoldLO** [comment]
I concur with the OP.  There is a lot of complexity to GitHub and the documentation is rather sparse in areas that a new dev could use to become more proficient.  There are a lot of samples but this does not help to learn context.  Copy/paste coding is dangerous in the wrong hands (i.e. the vibe coder, the new dev, or someone in a hurry).
